src/utils/companyNumberSanitizer.ts

```diff
@@ -0,0 +1,37 @@
+import { provide } from 'inversify-binding-decorators'
+
+@provide(CompanyNumberSanitizer)
+export default class CompanyNumberSanitizer {
+  private readonly COMPANY_NUMBER_SIZE: number = 8
+
+  public sanitizeCompany(companyNumber: string): string {
+    if (!companyNumber) {
+      throw new Error('Company number is required')
+    }
+
+    companyNumber = companyNumber.toUpperCase()
+    companyNumber = this.stripWhitespaces(companyNumber)
+
+    return this.padNumber(companyNumber)
+  }
+
+  private stripWhitespaces(companyNumber: string): string {
+    return companyNumber.replace(/\s/g, '')
+  }
+
+  private padNumber(companyNumber: string): string {
+    if (/^([a-zA-Z]{2}?)/gm.test(companyNumber)) {
```

Comment:
Consider moving this regex to a constant at the top of this file. Maybe move this out to a private method to be more descriptive too i.e. `isAlphaPrefix`